### PR TITLE
Add a gallery example for varying transparent points

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Add a gallery example for the new feature in #626.
image

Address #615. 

Preview: https://pygmt-qmth2blje.vercel.app/gallery/plot/points-transparency.html

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [x] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

**Comment by seisman** [comment]
Ping @weiji14 for a review.

**Comment by weiji14** [comment]
Sorry, a bit busy right now, so a quick comment instead of a full review. Could we use a simpler straight line example instead of a sine curve here? I.e. like this image in our unit test (but maybe modify it slightly)

circle transparency

**Comment by seisman** [comment]
The example is rewritten to plot the figure below. 
image

File: examples/gallery/plot/points-transparency.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""
+Points with varying transparency
+--------------------------------
+
+Plotting points with varying transparency is simply passing an array to the
+``transparency`` argument.
```
Comment (weiji14) [inline examples/gallery/plot/points-transparency.py:6]:
```suggestion
Points can be plotted with different transparency levels by passing in an array to the
``transparency`` argument of :meth:`pygmt.Figure.plot`.
```

**Review by weiji14 — APPROVED** [review]

**Pull request merged**